applications/vanilla/models/class.categorymodel.php

```diff
@@ -93,6 +99,10 @@ public function __construct() {
         parent::__construct('Category');
         $this->collection = $this->createCollection();
         $this->eventManager = Gdn::getContainer()->get(EventManager::class);
+
+        $container = Gdn::getContainer();
+        $this->breadcrumbModel = $container->get(BreadcrumbModel::class);
+        $this->breadcrumbModel->addProvider($container->get(ForumBreadcrumbProvider::class));
```

Comment:
I think Vanilla has [a hook that's already adding `ForumBreadcrumbProvider` to instances of  `BreadcrumbModel`](https://github.com/vanilla/vanilla/blob/ccbcaceb7343d3ce9986d9ed5127fe9eeaf7a327/applications/vanilla/settings/class.hooks.php#L26) that are instantiated by the container. Are you sure you need to add the provider it here too?

Reply:
Yes, for some reason `container` returns empty list of providers for BreadcrumbModel here. I think that is some legacy incompatibility. As you can see we do not receive container in constructor as a param. And when I try to add anything there - it is not passed. 
I don't want to do "yak shaving" here. This is just a "patch". If you have an idea how I can get correctly instantiated BreadcrumbModel with all providers I would appreciate any help.

Reply:
@initvector I had discussion with @charrondev , and we've found the issue. 
I moved initializing of BreadcrumbModel out of constructor, it seem better now. 
Please check updated.